### DISCUSSION TITLE
Move all phpUtils to a composer package

### DISCUSSION THREAD

**Discussion by webrgp** [Shifu462/twiggy#48] [Ideas]
Different frameworks have different ways to setup twig, som it makes sense to move that to different composer packages for each framework (Symphony, Craft CMS, etc).

Then Twiggy can look at specific paths, either by overriding configuration, or by checking the project's `composer.json` to decide.

In PHPStorm + Craft land, this is done through the [craft-autocomplete](https://github.com/nystudio107/craft-autocomplete) package. I think it makes sense and will simplify the Twiggy extension quite a bit.

**Comment by webrgp** [comment]
This could also solve the https://github.com/moetelo/twiggy/issues/23 issue with docker setups.

**Reply by moetelo** [reply]
I don’t quite understand the idea. How exactly would this approach solve it?

The difference is that `craft-autocomplete` is entirely independent of everything else. You run `php craft autocomplete/generate` once, and it generates files that enhance autocomplete features.

In contrast, Twiggy runs `php phpUtils/doSomething.php` each time it needs to make a request within the PHP environment.

**Reply by webrgp** [reply]
My limited understanding of Twiggy's phpUtils is that it returns JSON, that in turn is picked up by the plugin to perform its various tasks.

What if a composer package could do the same, and the plugin only needed to run `[php] ./vendor/twiggy/craft-autocomplete` or `[php] ./vendor/twiggy/symphony-autocomplete`, where `[php]` is the plugin's configurable option that could be replace with `ddev php` or something else?

My initial idea was to try to retrieve the data we get from the JSON through PHP static analysis, but I realized that VSCode PHP doesn't do that out of the box. I have [PHP Intelephense](https://github.com/bmewburn/vscode-intelephense) installed for that.

**Reply by moetelo** [reply]
Ah, I see. It's based on the assumption that ./vendor will be available in the PHP docker container without any additional tweaks (which is quite correct), comparing to some /home/.../.vscode/extensions/twiggy directory.
